src/runtime/pyobject.cs

```diff
@@ -119,7 +119,9 @@ public static PyObject FromManagedObject(object ob)
                 Runtime.XIncref(Runtime.PyNone);
                 return new PyObject(Runtime.PyNone);
             }
-            IntPtr op = CLRObject.GetInstHandle(ob);
+            IntPtr op = typeof(Exception).IsAssignableFrom(ob.GetType()) ?
+                Exceptions.GetExceptHandle((Exception)ob)
+                : CLRObject.GetInstHandle(ob);
```

Comment:
I feel like `Exceptions.GetExceptionHandle` should be built into `CLRObject.GetInstHandle`.

Reply:
The exception object is a special case, we should keep it outside from the generic case `GetInstHandle`, for example, if it has multiple special cases instead of one, you won't want to contaminate the CLRObject methods, right? Just like I removed the special case code from the CLRObject's constructor.